core-api/src/test/java/com/optimizely/ab/config/ProjectConfigTest.java

```diff
@@ -345,4 +358,53 @@ public void setForcedVariationMultipleUsers() {
         assertNull(projectConfig.getForcedVariation("etag2", "testUser2"));
 
     }
+
+    @Test
+    public void getAttributeIDWhenAttributeKeyIsEqualtoBotFiltering() {
+        ProjectConfig projectConfig = ProjectConfigTestUtils.validProjectConfigV4();
+        String attributeID = projectConfig.getAttributeId(projectConfig, ReservedAttributeKey.BOT_FILTERING_ATTRIBUTE.toString());
```

Comment:
You should hardcode '$opt_bot_filtering' here. Using same enum for comparison both in the code and test leave room for error if in case there is a typo when defining enum.